### PR TITLE
dbeaver/dbeaver#21574 eclipse configuration directory in your profile

### PR DESCRIPTION
**Pull request by serjiokov**
The configuration options specified in order to store data in OS specific folders.

### CODE REVIEW

File: product/community/DBeaver.product [hunk 1]
```diff
@@ -94,6 +94,9 @@
         <plugin id="org.eclipse.equinox.event" autoStart="true" startLevel="2"/>
         <!-- This is important because simpleconfigurator conflicts with update.configurator -->
         <plugin id="org.eclipse.update.configurator" autoStart="false" startLevel="10"/>
```
Comment (serge-rider) [inline product/community/DBeaver.product]: conf -> "install-data"

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: product/community/DBeaver.product [hunk 2]
```diff
@@ -94,6 +94,9 @@
...
+        <property name="osgi.configuration.area" value="@user.home/AppData/Roaming/DBeaverData/conf" os="win32" />
+        <property name="osgi.configuration.area" value="@user.home/.config/DBeaverData/conf" os="linux" />
```
Comment (serge-rider) [inline product/community/DBeaver.product]: Not sure about Linux path. It has to be in $XDG_DATA_HOME

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: product/community/DBeaver.product (hunk 2 above)
Comment (serjiokov) [reply product/community/DBeaver.product]: on OS the $XDX_DATA_HOME have to be resolved into 

image

File: product/community/DBeaver.product (hunk 1 above)
Comment (serjiokov) [reply product/community/DBeaver.product]: done

**Comment by serjiokov** [comment]
For now reached next points:
1. Organized own dbeaver launcher with implemetation of store relative configurtaion (../DBeaverData/install-data) in OS specific location:
    a. WIN - APPDATA
    b. UNIX - XDG_DATA_HOME 
In order to achive it next properties in product require:
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="win32" />
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="linux" />
        <property name="osgi.configuration.area" value="**@data.home**/DBeaverData/install-data" os="macosx" />

Not achived:
Generation .ini file option (or phaze of product materialization) still paste 
"-startup 
  plugins/org.eclipse.equinox.launcher_1.6.500.v20230717-2134.jar" that does not fit requiremnts.
This line have to be changed to 
" -startup
plugins/org.jkiss.dbeaver.launcher_[version]jar" 

Result:
Product configuration data will be stored at `"~/AppData/DBeaverData/install-data"`
The "~/.eclipse" not used any more.

File: plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
+/*
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java]: Can we just import all this stuff from original launcher and not copy-paste it?
We have it in plugin list anyway.

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java]: I don't think so we targeted to reimplement a private method 
` private URL buildLocation(String property, URL defaultLocation, String userDefaultAppendage) `

Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java]: Yes. But why can't we just import constants, JNI bridge and other stuff which is simply copy-pasted?

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Constants.java]: Done, but one thing the equinox classes is in internal package.

**Comment by Matvey16** [comment]
Verified

**Comment by Matvey16** [comment]
Verified

**Pull request merged**